redbot/cogs/economy/economy.py

```diff
@@ -255,7 +260,12 @@ def __init__(self, bot: Red):
         else:
             next_payday = await self.config.member(author).next_payday()
             if cur_time >= next_payday:
-                await bank.deposit_credits(author, await self.config.guild(guild).PAYDAY_CREDITS())
+                credit_amount = await self.config.guild(guild).PAYDAY_CREDITS()
+                for role in author.roles:
```

Comment:
why iterate through all of a user's roles to find a payday amount rather than just getting the payday amount based on the user's top role?

Reply:
All roles don't necessarily have a corresponding pay-out, so while their top role might be Admin, their highest pay-out role might be Gold/Platinum or the like.